src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/Abstractions/IMacro.cs

```diff
@@ -31,6 +31,6 @@ public interface IMacro<T> : IMacro where T : IMacroConfig
         /// <summary>
         /// Evaluates the macro based on <paramref name="config"/>. The result is modification of variable collection <paramref name="variables"/>.
         /// </summary>
-        void Evaluate(IEngineEnvironmentSettings environmentSettings, IVariableCollection variables, T config);
+        void Evaluate(IEngineEnvironmentSettings environmentSettings, IVariableCollectionEx variables, T config);
```

Comment:
At the moment only `Coalesce` needs it. 
We can probably just try to cast to `IVariableCollectionEx` in `CoalesceMacro`. 
But I'm not too strong on it.